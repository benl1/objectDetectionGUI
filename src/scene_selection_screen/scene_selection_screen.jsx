import { displayImageChoiceScreen, displayOutputScreen } from '../entrypoint';
import { displayYesNoDialog, displayImageUploadDialog, displayErrorDialog } from '../control/dialogs';
import React from 'react';

export default class SceneSelectionScreen extends React.Component {
    render() {
        return (
            <div>
                <SceneSelectionScreenTitle />
                <SceneSelectionContainer app={this.props.app}/>
                <div className='button' onClick={() => displayImageChoiceScreen(this.props.app)}>Image choice</div>
            </div>
        );
    }
}

class SceneSelectionContainer extends React.Component {
    handleSceneImageUpload() {
        const img_paths = displayImageUploadDialog();
        if (img_paths === undefined) return;

        const scene_img_path = img_paths[0];
        console.log(`set the scene image: ${scene_img_path}`);
        
        displayOutputScreen(this.props.app, {
            input: 'scene_image',
            path: scene_img_path
        });
    }

    handleWebcam() {
        /* first, check to make sure that the user has a webcam plugged in. 
         * if they have more than one, let them choose which device they'd like to use
         * along with some properties, such as the frame-rate, image height and width, etc.
         */
        // list all the media devices (audio in, out, video in, screen-capture, etc.)
        // for now, i'm just passing the first media device as the one that's been selected
        // along with some basic properties such as height and width
        navigator.mediaDevices.enumerateDevices()
            .then(devices => {
                const video_devices = devices.filter(device => (device.kind === 'videoinput'));
                console.log(video_devices);

                if (video_devices.length == 0) {
                    displayErrorDialog('no webcams found!');
                } else {
                    displayOutputScreen(this.props.app, {
                        input: 'webcam', 
                        id: video_devices[0].deviceId, 
                        media_options: {video: true, audio: false}
                    });
                }
            })
            .catch(err => console.error(err));
    }

    render() {
        return (
            <div>
                <div className='button' onClick={() => this.handleSceneImageUpload()}>
                    Upload scene image
                </div>
                <div className='button' onClick={() => this.handleWebcam()}>
                    Use a webcam
                </div>
            </div>
        );
    }
}

function SceneSelectionScreenTitle(props) {
    return <h3>Select scene imagery source</h3>;
}