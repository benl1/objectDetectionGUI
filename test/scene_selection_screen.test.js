// image_choice_screen.test.js
import React from 'react';
import { shallow, mount } from 'enzyme';
import { SceneSelectionScreen, SceneSelectionContainer, PictureTaker } from '../src/scene_selection_screen/scene_selection_screen';
import renderer from 'react-test-renderer';
import App from '../src/control/app';

describe('Image Choice screen', () => {

    it('test that scene selection renders correctly', ()=>{
        const component = renderer.create(
            <SceneSelectionScreen app={new App()} />,
            document.getElementById('root')
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('test that scene selection container renders correctly', ()=>{
        const component = renderer.create(
            <SceneSelectionContainer app={new App()} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    // it('test that PictureTaker renders correctly', ()=>{
    //     const component = renderer.create(
    //         <PictureTaker app={new App()} />
    //     );
    //     let tree = component.toJSON();
    //     expect(tree).toMatchSnapshot();
    // })

    it('test that scene selection container renders PictureTaker when the state is changed', ()=>{        
        let component = shallow( <SceneSelectionContainer app={new App()} />);
        expect(component.find(PictureTaker)).toHaveLength(0);        
        component.instance().setState({showPictureTaker: true});
        component.instance().forceUpdate();
        component = component.update();
        expect(component.find(PictureTaker)).toHaveLength(1);
    })

});