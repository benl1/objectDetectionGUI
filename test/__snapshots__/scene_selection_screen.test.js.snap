// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scene Selection screen test that PictureTaker renders correctly 1`] = `
<div
  className="flexColumn"
>
  <WebcamOutput
    height={500}
    width={500}
  />
  <div
    className="button"
    onClick={[Function]}
  >
     Grab Frame 
  </div>
</div>
`;

exports[`Scene Selection screen test that scene selection container renders correctly 1`] = `
<div
  className="flexColumn"
>
  <div
    className="button"
    onClick={[Function]}
  >
    Upload scene image
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    Take a picture
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    Take a video
  </div>
</div>
`;

exports[`Scene Selection screen test that scene selection renders correctly 1`] = `
<div
  className="flexColumn"
>
  <h3>
    Select scene imagery source
  </h3>
  <div
    className="flexColumn"
  >
    <div
      className="button"
      onClick={[Function]}
    >
      Upload scene image
    </div>
    <div
      className="button"
      onClick={[Function]}
    >
      Take a picture
    </div>
    <div
      className="button"
      onClick={[Function]}
    >
      Take a video
    </div>
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    Image choice
  </div>
</div>
`;
